src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```
@@ -0,0 +1,182 @@
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Usage,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        private const string RequiresPreviewFeaturesAttribute = nameof(RequiresPreviewFeaturesAttribute);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private bool thisAssemblyUsesPreviewFeatures;
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                IAssemblySymbol? thisAssembly = context.Compilation.Assembly;
+                ImmutableArray<AttributeData> attributes = thisAssembly.GetAttributes();
+
+                AttributeData? attribute = attributes.FirstOrDefault(x => x.AttributeClass.ToString() == RequiresPreviewFeaturesAttribute);
```

Should probably be symbol comparison instead.

---

In CA1416 we used string comparison with attribute name so that it could be used in environments lower than 5.0. I guess this attribute would newer be used in lower envs as it is about a new feature, so probably better to load the type from the compilation, which also would allow failing fast if the symbol is not found 
```cs
if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeVersioningRequiresPreviewFeaturesAttribute, out var previewFeaturesAttribute))
{
    return;
}
```